### ISSUE TITLE
lbonn/rofi window squeeze after hyprland update

### ISSUE THREAD

**Issue by Aleksanaa** [hyprwm/Hyprland#1484]
After upgrading hyprland (can reproduce in latest commit bd2245d642f700ac736cd8349352917ef5340dbc), lbonn/rofi 1.7.5-wayland1 window squeeze with fixed height disabled. Not sure whether other programs using gtk-layer-shell and adaptive width/height have the same problem.

Quick test:
```
rofi -show drun -theme-str 'listview { fixed-height: false; }' -no-config
```

Also opened an issue there: https://github.com/lbonn/rofi/issues/75

**Comment by vaxerski** [comment]
cna you do a git bisect to narrow down the commit?

**Comment by vaxerski** [comment]
should be fixed with a431c1b01c08cdc1c2f030bebf052aa4f534d12a, please check

**Comment by Aleksanaa** [comment]
> should be fixed with [a431c1b](https://github.com/hyprwm/Hyprland/commit/a431c1b01c08cdc1c2f030bebf052aa4f534d12a), please check

Yes, this seems partially solve the issue. When changing the first time it is usually correct but the second is wrong, and the window moves upwards. (Overflow?)

![image](https://user-images.githubusercontent.com/42209822/216776854-09975e17-69fe-4d69-ba52-72a7bf7b3792.png)

**Comment by Aleksanaa** [comment]
And sometimes it corrects itself after a few seconds.

**Comment by MonstrousOgre** [comment]
I'm facing a similar issue but for me, rofi is a lot bigger than it's supposed to be. It fixes itself after less than a second though, so it's usually not noticeable. I don't know exactly which commit is causing this, but it was working a few days ago. I'll try to pinpoint this.

**Comment by Aleksanaa** [comment]
> I'm facing a similar issue but for me, rofi is a lot bigger than it's supposed to be. It fixes itself after less than a second though, so it's usually not noticeable. I don't know exactly which commit is causing this, but it was working a few days ago. I'll try to pinpoint this.

Not the same one, but I think the different behaviour may be caused by different fractional scaling values.

**Comment by MonstrousOgre** [comment]
> Not the same one, but I think the different behaviour may be caused by different fractional scaling values.

I thought fractional scaling wasn't implemented in Hyprland yet?

So turns out fnott (my notification daemon) is also having this problem. Except the scaling and position are completely random in fnott's case. Sometime's it's too big and other times it's too small.

Since I'm travelling, I won't be able to check which commit is causing this until Friday. I installed fnott on Feb 3rd, and it was working fine then. Even now it's working fine if I use v0.21.0beta instead of the latest commit

**Comment by Aleksanaa** [comment]
> I'm facing a similar issue but for me, rofi is a lot bigger than it's supposed to be. It fixes itself after less than a second though, so it's usually not noticeable. I don't know exactly which commit is causing this, but it was working a few days ago. I'll try to pinpoint this.

Found that this is also happening on my setup but I didn't discover

**Comment by Aleksanaa** [comment]
So I think this indicates ef12120270c6a16a7680a20ded3899140eee48d0 is still not finished yet. I'll stick to v0.21.0beta for a while.

**Comment by vaxerski** [comment]
fixed with cb6c47098d32fe937ff8f6abce7963354dca0799

I hate all of you.

Still seems like a protocol violation to me from rofi's side. Rofi sends bullshit height values for its status here:

https://github.com/lbonn/rofi/blob/5d4a3e14788945a9a1c7d18993151a3b00649373/source/wayland/view.c#L147-L158

The height reported is almost always wrong and does not match the buffer.
